src/shared/copy.c

```diff
@@ -961,33 +962,22 @@ static int fd_copy_directory(
         if (!d)
                 return -errno;
 
-        exists = false;
-        if (copy_flags & COPY_MERGE_EMPTY) {
-                r = dir_is_empty_at(dt, to, /* ignore_hidden_or_backup= */ false);
-                if (r < 0 && r != -ENOENT)
-                        return r;
-                else if (r == 1)
-                        exists = true;
-        }
+        r = dir_is_empty_at(dt, to, /* ignore_hidden_or_backup= */ false);
+        if ((r > 0 && !(copy_flags & (COPY_MERGE|COPY_MERGE_EMPTY))) || (r == 0 && !FLAGS_SET(copy_flags, COPY_MERGE)))
+                return -EEXIST;
+        if (r < 0 && r != -ENOENT)
+                return r;
```

Comment:
we usually do error checks first, hence simply swap the two if checks?

why not like this:

```c
if (r < 0) {
        if (r != -ENOENT)
                return r;
        exists = false;
} else {
        if `(!(copy_flags & (COPY_MERGE|COPY_MERGE_EMPTY) || (r == 0 && !FLAGS_SET(copy_flags, COPY_MERGE))
                return -EEXIST;

        exists = true;
}
```
that way there's no redundant check

Reply:
ah, i see now that the exists boolean is still needed firther down